tumblr-photo-video-ripper.py

```diff
@@ -206,19 +206,45 @@ def _download_media(self, site, medium_type, start):
                         # if post has photoset, walk into photoset for each photo
                         photoset = post["photoset"]["photo"]
                         for photo in photoset:
-                            self.queue.put((medium_type, photo, target_folder))
+                            self.queue.put((site, medium_type, photo, target_folder))
                     except:
                         # select the largest resolution
                         # usually in the first element
-                        self.queue.put((medium_type, post, target_folder))
+                        self.queue.put((site, medium_type, post, target_folder))
                 start += MEDIA_NUM
-            except KeyError:
-                break
+                retry_times = 0
+                retry_times_for_site = 0
+            except (KeyboardInterrupt, SystemExit):
+                raise
             except UnicodeDecodeError:
-                print("Cannot decode response data from URL %s" % media_url)
+                try:
+                    print("For site %s, Cannot decode response data from URL %s" % (site, media_url))
+                except:
+                    print("For site %s, Cannot decode response data from URL" % site)
+                if retry_times > RETRY:
```

Comment:
Duplicate codes here. We only need to have this retry logic in one place.

Reply:
Moved the retry logic to a single function.